### PR TITLE
Trigger single task or reference entity execution

### PR DESCRIPTION
**Pull request by katrogan**

### CODE REVIEW

File: flytekit/remote/execution.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    auth_role: AuthRole = None
+    # optional name to assign to execution
+    name: str = None
+    # The name of the user launching this execution.
+    principal: str = None
+
+
+def execute(
+    entity: Union[ReferenceEntity, ContainerTask],
+    client: SynchronousFlyteClient,
```
Comment (kumare3) [inline flytekit/remote/execution.py:35]: can this be optional?

File: flytekit/remote/execution.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            try:
+                client.create_task(task_identifer=sdk_task.id, task_spec=TaskSpec(sdk_task))
+            except user_exceptions.FlyteEntityAlreadyExistsException:
+                pass
+    else:
+        reference_identifier = entity.id
+
+    with flyte_context.FlyteContext.current_context():
+        literals = translate_inputs_to_literals(
+            ctx, input_kwargs=inputs, interface=entity.interface, native_input_types=entity.get_input_types()
```
Comment (kumare3) [inline flytekit/remote/execution.py:61]: the referenceEntity may not have an interface right? so we should fetch that first?

Comment (katrogan) [reply flytekit/remote/execution.py:61]: oh I did not know this! good call, yes

File: flytekit/remote/execution.py (hunk 1 above)
Comment (katrogan) [reply flytekit/remote/execution.py:35]: sure, we can always initialize using the platform url in the config

File: flytekit/remote/execution.py (hunk 2 above)
Comment (katrogan) [reply flytekit/remote/execution.py:61]: actually it looks like inputs and outputs are "required" input arguments for reference entities but i will fetch if they are none

**Comment by katrogan** [comment]
PTAL @kumare3

File: flytekit/remote/execution.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    name: str = None
+    # The name of the user launching this execution.
+    principal: str = None
+
+
+def execute(
+    entity: Union[Identifier, ContainerTask],
+    project: str,
+    domain: str,
+    inputs: Dict[str, Any] = None,
```
Comment (kumare3) [inline flytekit/remote/execution.py:39]: should we inputs as a dictionary or just kwargs? Whats your gut feeling?

Comment (katrogan) [reply flytekit/remote/execution.py:39]: kwargs gives us consistency with just calling the function in local execute mode but b/c there are required/positional args in this scenario naming inputs isn't what departs from that pattern. what do you think? happy to change or keep as is

**Pull request closed without merging**